js/deleteTask.js

```diff
@@ -0,0 +1,19 @@
+import { taskLIST } from "/js/addToTask.js";
+import { CLICK_EVENT, DELET, UL_LIST } from "/js/constants.js";
+
+function removeToDos(element) {
+  let index = taskLIST.findIndex((item) => {
+    return item.id == element.id;
+  });
+  taskLIST[index].trash = true;
+  element.parentNode.parentNode.removeChild(element.parentNode);
+  taskLIST.splice(index, 1);
```

Comment:
splice() shifts all the elements after the deleted index down by one position, don't you think this is making the code a bit expensive for large arrays, rather than using splice can we do anything else here to delete the item? Do you have anything else on your mind?

Reply:
I can use Array.filter() to create an array that doesn't contain the element which I want to delete. but frequently creating new array is memory consuming also. Next I can think of linked list only. Should I implement linked list to add, remove, update the taskList? if I am missing any more functionalities to do so, please share.

Reply:
Thank you for considering my suggestion and updating the code. Using Array.filter() to remove an element seems like a good alternative to splice() for avoiding the performance impact on large arrays. I understand that creating a new array can also have memory implications, but it's great that you're exploring different options for optimization.

Regarding linked lists, they can be a good choice for scenarios where frequent insertions and deletions are needed, since they don't require shifting elements like arrays do. However, implementing a linked list can also add complexity and may not be necessary for the current use case. Let's keep the code like it is right now and let's see what our Mentors think of it/ wait for their reviews. Thanks